gcc-16: rc.c:65: warning: variable ‘ln’ set but not used

Hi,

This is somewhat trivial, but a build with gcc-16 shows up a warning which breaks builds using `-Werror`:

```
gcc-16 -DLOCALEDIR=\"/usr/share/locale\" -DHAVE_CONFIG_H   -I/usr/include/x86_64-linux-gnu -isystem /usr/include/mit-krb5 -isystem /usr/include/mit-krb5 -I/usr/include/p11-kit-1 -I/usr/include/libxml2 -I/usr/include/libsecret-1 -I/usr/include/gio-unix-2.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/sysprof-6 -pthread  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/home/andy/git/mcds=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wall -pedantic -Werror -Wno-error=date-time -c -o mcds-rc.o src/rc.c
src/rc.c: In function ‘read_rc’:
src/rc.c:65:13: error: variable ‘ln’ set but not used [-Werror=unused-but-set-variable=]
   65 |         int ln = 0;                    /* Line number */
      |             ^~
cc1: all warnings being treated as errors
```

There's a line number counter that doesn't get used. Do you have a preference about how to deal with it? Perhaps might as well remove it as it would be easy to add back in if it were ever wanted?

Disabling the warning or using `[[maybe_unused]]` or equivalent are alternative approaches of course.

Thanks!

--- a/src/rc.c
+++ b/src/rc.c
@@ -62,7 +62,6 @@ read_rc(const char *file)
 {
 
 	int i  = 0;                    /* Temporary loop indexer */
-	int ln = 0;                    /* Line number */
 	int len = 0;                   /* String length */
 	char *home = NULL;             /* Home directory */
 	char *pfile = NULL;            /* Password file */
@@ -116,7 +115,6 @@ read_rc(const char *file)
 	}
 
 	while (fgets(line, LINE_MAX, ifd) != NULL) {
-		++ln;
 		lptr = line;
 		i = 0;
 		if (line[0] != '\n' && line[0] != '#') {

